app/sidekiq/evss/disability_compensation_form/upload_bdd_instructions.rb

```diff
@@ -88,20 +114,35 @@ def perform(submission_id)
 
       private
 
+      def file_body
+        @file_body ||= File.read('lib/evss/disability_compensation_form/bdd_instructions.pdf')
+      end
+
       def upload_bdd_instructions
-        client.upload(file_body, document_data)
+        # NOTE: the ApiProviderFactory implements its own Flipper,
+        # ApiProviderFactory::FEATURE_TOGGLE_UPLOAD_BDD_INSTRUCTIONS, which will be used to iteratively
+        # start sending more BDD Instructions to Lighthouse instead of EVSS
+        # However, since using this factory is new behavior, it is hidden behind this additional Flipper, which will act
+        # as a "kill switch" in the event there are problems with the ApiProviderFactory implementation, so we can
+        # revert back to using the EVSS::DocumentsService directly here
+        if Flipper.enabled?(:disability_compensation_use_api_provider_for_bdd_instructions)
+          upload_via_api_provider
+
+          upload_provider.log_upload_success(STATSD_KEY_PREFIX)
+        else
+          EVSS::DocumentsService.new(submission.auth_headers).upload(file_body, document_data)
+        end
       end
 
-      def file_body
-        @file_body ||= File.read('lib/evss/disability_compensation_form/bdd_instructions.pdf')
+      def upload_via_api_provider
+        document = upload_provider.generate_upload_document(
+          BDD_INSTRUCTIONS_DOCUMENT_TYPE, BDD_INSTRUCTIONS_DOCUMENT_TYPE
+        )
+        upload_provider.submit_upload_document(document, file_body)
       end
 
-      def client
-        @client ||= if Flipper.enabled?(:disability_compensation_lighthouse_document_service_provider)
-                      # TODO: create client from lighthouse document service
-                    else
-                      EVSS::DocumentsService.new(submission.auth_headers)
-                    end
+      def upload_provider
+        @upload_provider ||= EVSS::DisabilityCompensationForm::UploadBddInstructions.api_upload_provider(submission)
```

Comment:
I might not being seeing the whole picture here but is this only called on the EVSS path? Or why is there not a flipper here too? I see there is the `||=` so assume the flow/logic means this is set already to the api provider in normal cases, but having trouble working out the flow to support that.

Reply:
The use of this is behind the `disability_compensation_use_api_provider_for_bdd_instructions` flipper I used to hide all new functionality in this job on [this line](https://github.com/department-of-veterans-affairs/vets-api/pull/18347/files#diff-2e1a8502612bbe5107bee6b35d8bec26487d7846b228614f9c4f28cde40cfac7R128).

I can see this being confusing though, I had to use a class method [here](https://github.com/department-of-veterans-affairs/vets-api/pull/18347/files#diff-2e1a8502612bbe5107bee6b35d8bec26487d7846b228614f9c4f28cde40cfac7R83) for getting the provider from the API Provider factory, because I'm also using it for my logging in the `sidekiq_retries_exhausted` block and I don't have access to the instance there.

The method you commented on here is the instance method that calls the class method so I can access the provider in the job instance; and because this job is namespaced under EVSS it makes this line even more confusing.

If it's easier we can look at this together there may be another way to organize this so it's more self documenting but it's kind of hard to describe the issues I ran into with this job in the PR description. What do you think?